service/controller/resource/deployment/resource.go

```diff
@@ -131,6 +130,18 @@ func isDeploymentModified(a, b *v1.Deployment) bool {
 	return false
```

Comment:
You should just return `aVersion != bVersion` here and drop the `if` block above.

Reply:
Updated